### ISSUE TITLE
Feature Request: Immediate clear

### ISSUE THREAD

**Issue by McPo** [undefinederror/vue-lazy#3]
Would be handy to ignore debouncing if the (x) button on an input field is clicked. (As I've noticed other debouncing libraries provide options for this)

**Comment by undefinederror** [comment]
Hi @McPo sorry the late reply I didn't see the issue.
It's a good suggestion I'll see what I can do.
Have you got a link to any of those debouncing libraries?

**Comment by McPo** [comment]
https://github.com/dhershman1/vue-debounce has a `fireonempty` option

(Although when I made this FR I think I may have been referring to a different library all together, however Im unable to recall which one). I'll let you know If I come across it again. (I should have listed it in the initial FR)

**Comment by undefinederror** [comment]
Hey @McPo ,

Well that's quite a nice library! It has a good bunch of options in addition to the `fireonempty`

Now, the `fireonempty` is a bit too specific for the scope of this library, which is relatively simple and generic, but to address your request I have added a third type, `sync`, which effectively makes the input event fire immediately.

I suppose you can leverage that with a bit of logic in a computed prop, like

```
export default {
  data(){
    return {
      val:'yo'
    }
  },
  computed:{
    type(){
      !this.val ? 'sync' : 'debounce'
    }
  }
}
```
and in your template
`<input type="text" v-model.trim="val"  v-lazy-input:[type]/>`

Although I am not sure how that would apply since once the input receives the first character it will start debouncing again...

Anyway, there you have it
Hope it works for you
It's published on npm as 1.1.0

**Comment by undefinederror** [comment]
I am going to close this since as far as I am concerned the issue is resolved
